pyxel/core/src/pyxelcore/graphics.cc

```diff
@@ -254,6 +254,66 @@ void Graphics::DrawCircleBorder(int32_t x,
   }
 }
 
+void Graphics::DrawTriangle(int32_t x1, int32_t y1, int32_t x2, int32_t y2, int32_t x3, int32_t y3, int32_t color){
+  int32_t draw_color = GET_DRAW_COLOR(color);
+
+  //rank as y3 > y2 > y1
+  if(y1 > y2){
+    std::swap(y1, y2);
+    std::swap(x1, x2);
+  }
+  if(y1 > y3){
+    std::swap(y1, y3);
+    std::swap(x1, x3);
+  }
+  if(y2 > y3){
+    std::swap(y2, y3);
+    std::swap(x2, x3);
+  }
+  // slide bottom-up from y1 to y3
+  float alpha12 = (y2==y1) ? 0 : static_cast<float>(x2 - x1) / (y2 - y1);
+  float alpha13 = (y3==y1) ? 0 : static_cast<float>(x3 - x1) / (y3 - y1);
+  float alpha23 = (y3==y2) ? 0 : static_cast<float>(x3 - x2) / (y3 - y2);
+  int32_t x_intersection = x1 + alpha13 * (y2 - y1)  + 0.5f;
+  int32_t y_slider = y1;
+  for(; y_slider <= y2; y_slider++){
+      int32_t x_slider, x_end;
+
+      if(x_intersection < x2){
+        x_slider = x_intersection + alpha13 * (y_slider - y2) + 0.5f;
```

Comment:
compute `x_slider` and `x_end` using `x_intersection`, could avoid the case that y1==y2 and y2==y3